python/ray/tune/trial_scheduler.py

```diff
@@ -47,11 +48,18 @@ def on_trial_remove(self, trial_runner, trial):
 
         raise NotImplementedError
 
-    def choose_trial_to_run(self, trial_runner, trials):
+    def track_experiment(self, experiment, trial_runner):
+        """Tracks experiment from which trials and generated and queued."""
```

Comment:
"""Adds an experiment to the scheduler. The scheduler is responsible for adding the trials of the experiment to the runner, which can be done immediately (if there are a finite set of trials), or over time (if there is an infinite stream of trials)."""